### PR TITLE
Add new versions of giflib

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/giflib/package.py [hunk 1]
```diff
@@ -11,11 +11,34 @@ class Giflib(AutotoolsPackage, SourceforgePackage):
...
+    def autoreconf(self, spec, prefix):
+        return
+
+    @when('@5.2.0:')
+    def install(self, spec, prefix):
+        make('install', 'PREFIX={0}'.format(prefix))
+
+    def patch(self):
+        if self.spec.satisfies('@5.2.0:'):
+            touch(join_path(self.stage.source_path, 'configure'))
```
Comment (haampie) [inline var/spack/repos/builtin/packages/giflib/package.py]: Is this to make the autotools package happy in the sense that it needs a configure file?

Comment (iarspider) [reply var/spack/repos/builtin/packages/giflib/package.py]: Exactly

File: var/spack/repos/builtin/packages/giflib/package.py [hunk 2]
```diff
@@ -11,11 +11,34 @@ class Giflib(AutotoolsPackage, SourceforgePackage):
...
 
-    version('5.1.4', sha256='df27ec3ff24671f80b29e6ab1c4971059c14ac3db95406884fc26574631ba8d5')
+    version('5.2.1', sha256='31da5562f44c5f15d63340a09a4fd62b48c45620cd302f77a6d9acf0077879bd')
+    version('5.2.0', sha256='dc7257487c767137602d86c17098ec97065a718ff568a61cfcf1a9466f197b1f')
+    version('5.1.4', sha256='df27ec3ff24671f80b29e6ab1c4971059c14ac3db95406884fc26574631ba8d5', extension='tar.bz2')
+
+    depends_on('automake', type='build')
+    depends_on('autoconf', type='build')
+    depends_on('m4', type='build')
+    depends_on('libtool', type='build')
```
Comment (haampie) [inline var/spack/repos/builtin/packages/giflib/package.py]: Is this missing `when=@:5.1`?

Comment (iarspider) [reply var/spack/repos/builtin/packages/giflib/package.py]: ~~Good catch~~
Actually, `AutotoolsPackage` will complain if any of these dependencies are not specified.

Comment (haampie) [reply var/spack/repos/builtin/packages/giflib/package.py]: Hm, maybe just remove the AutotoolsPackage bit and make it either a MakefilePackage or a Package where you run

```
configure('--prefix', prefix)
make()
make('install')
```
for 5.1.x?

**Review by haampie — APPROVED** [review]
Thanks!

**Pull request merged**